python/python/lance/ray/sink.py

```diff
@@ -39,10 +41,27 @@ def _pd_to_arrow(
 
     if isinstance(df, dict):
         return pa.Table.from_pydict(df, schema=schema)
-    if _PANDAS_AVAILABLE and isinstance(df, pd.DataFrame):
+    elif _PANDAS_AVAILABLE and isinstance(df, pd.DataFrame):
         tbl = pa.Table.from_pandas(df, schema=schema)
-        new_schema = tbl.schema.remove_metadata()
-        new_table = tbl.replace_schema_metadata(new_schema.metadata)
+        tbl.schema = tbl.schema.remove_metadata()
+        return tbl
+    elif isinstance(df, pa.Table):
+        fields = df.schema.names
+        new_columns = []
+        new_fields = []
+        for field in fields:
+            col = df[field]
+            new_field = pa.field(field, col.type)
+            if (
+                pa.types.is_null(col.type)
+                and schema.field_by_name(field).type == pa.string()
```

Comment:
@Jay-ju @eddyxu Please correct me if I am wrong but this looks like `None` support is added only for the `pa.string` data type. I would assume that we want to `None` to apply to any `nullable` field regardless of data type.